statsmodels/multivariate/cancorr.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+
+"""Canonical correlation analysis
+
+author: Yichuan Liu
+"""
+from __future__ import division
+
+import numpy as np
+from numpy.linalg import svd
+import scipy
+import pandas as pd
+
+from statsmodels.base.model import Model
+from statsmodels.iolib import summary2
+from .multivariate_ols import multivariate_stats
+
+
+class CanCorr(Model):
+    """
+    Canonical correlation analysis using singluar value decomposition
+
+    For matrices exog=x and endog=y, find projections x_cancoef and y_cancoef
+    such that:
+        x1 = x * x_cancoef, x1' * x1 is identity matrix
+        y1 = y * y_cancoef, y1' * y1 is identity matrix
+    and the correlation between x1 and y1 is maximized.
+
+    Attributes
+    -----------
+    endog : array
+        See Parameters.
+    exog : array
+        See Parameters.
+    corr_values : array
+        The canonical correlation values
+    y_cancoeff: array
+        The canonical coeefficients for endog
+    x_cancoeff: array
+        The canonical coefficients for exog
+
+    References
+    ----------
+    .. [1] http://numerical.recipes/whp/notes/CanonCorrBySVD.pdf
+    .. [2] http://www.csun.edu/~ata20315/psy524/docs/Psy524%20Lecture%208%20CC.pdf
+    .. [3] http://www.mathematica-journal.com/2014/06/canonical-correlation-analysis/
+    """
+    def __init__(self, endog, exog, tolerance=1e-8, missing='none', hasconst=None, **kwargs):
+        super(CanCorr, self).__init__(endog, exog, **kwargs)
```

Comment:
I didn't read carefully enough before:
missing and hasconst keywords need to be forwarded to `super(...).__init__`